File: Towers_Of_Hanoi/Java/Hanoi.java
import java.util.*;
import java.lang.*;
import java.io.*;

public class Main
{
	//Towers of Hanoi problem algorithm implementation
    //See: http://en.wikipedia.org/wiki/Tower_of_Hanoi

	public static void move(int n, int src, int temp, int dest){

			if(n>=1){

				move(n-1,src,temp,dest);
				System.out.println("Move "+src+"->"+dest);
 				move( n - 1, temp, dest, src );

			}
		}


	public static void main (String[] args) throws java.lang.Exception{
    
     //testcases
     move(3,1,3,2);

	}
}

File: Towers_Of_Hanoi/C/sunilshahu/hanoi.c
/*
 * Towers of Hanoi problem algorithm implementation
 * See: http://en.wikipedia.org/wiki/Tower_of_Hanoi
 */
#include <stdio.h>

void move(no_of_disc, src_pole, dest_pole, temp_pole)
{
	if(no_of_disc >= 1) {
		move(no_of_disc - 1, src_pole, temp_pole, dest_pole);
		printf("Move %d -> %d\n", src_pole, dest_pole);
		move(no_of_disc - 1, temp_pole, dest_pole, src_pole);
	}
}

int main()
{
	move(3,1,3,2);
	return 0;
}
